pytensor/tensor/random/rewriting/basic.py

```diff
@@ -277,34 +269,35 @@ def is_nd_advanced_idx(idx, dtype):
         n_discarded_idxs = len(supp_indices)
         indices = indices[:-n_discarded_idxs]
 
-    # If no one else is using the underlying `RandomVariable`, then we can
-    # do this; otherwise, the graph would be internally inconsistent.
-    if is_rv_used_in_graph(rv, node, fgraph):
-        return False
-
     # Update the size to reflect the indexed dimensions
-    new_size = output_shape[: len(output_shape) - rv_op.ndim_supp]
+    if isinstance(size.type, NoneTypeT):
+        new_size = size
+    else:
+        shape_feature = getattr(fgraph, "shape_feature", None)
+        if not shape_feature:
+            return None
```

Comment:
Should this really return `None`? If yes, add a `# pragma: no cover` to ignore the missing line coverage.

Reply:
Yes, it should return None, we need the feature for this rewrite case to work properly